### DISCUSSION TITLE
Asking Info-Assistant to focus on a particular document in the chat

### DISCUSSION THREAD

**Discussion by michaeldeh** [microsoft/PubSec-Info-Assistant#485] [Q&A]
I'm looking for some advice on how I could effectively ground the model to focus its response on a particular source document from the collection of documents that have been uploaded.

For example, suppose I have 100 documents uploaded. I know what they're all named, and I want to ask the model about one particular document. I ask it to **_"summarise the source document named TGI.1001.pdf."_**

This doesn't work most of the time. If I modify the question to say something like **_"summarise content that is related to reference TGI.1001.pdf"_**, it sometimes works but more often than not will reference other sources incorrectly.

I've played around with the system prompt in the playground to add something like this:

> **_"...If the question includes a filename, then find that specific document and focus on it. Filenames will have the prefix of TGI or ABC, and end in .pdf. For example, TGI.12345.PDF."_**

If I then ask a question like **_"what can you tell me about the document with the name TGI.1001.pdf"_**, it yields slightly better results. But it still fails to consistently return documents, sometimes asserting they don't exist or referencing incorrect sources (though less so than the standard prompt).

**How could I go about directing the model in the chat to ground its answer with a specific source document based on its name?** Is it possible to do this by passing the `filename` (or some other parameter) in the question? Or is there another method?

**Other potentially relevant information**
- The filename is also consistently referenced in the top right corner of each document.
- In my use-case, the filename convention follows a predictable pattern (e.g., TGI.1001.pdf, TGI.1002.pdf, ...)
- The files vary in page length (some are a few pages, others are hundreds of pages)
- There are over 200 documents
- All documents are relatively homogenous in format and content and could be broadly categorised into three "types of content".
- The modified prompt described above struggles if I ask follow-up questions about _other_ documents, and is more accurate if I start a new chat for each new document.

Thanks!

**Comment by ArpitaisAn0maly** [comment]
Hello [michaeldeh](https://github.com/michaeldeh)

Generative AI responses are not deterministic. So using prompting to focus on one document may not work under all scenarios.  With IA you have option to use folder to focus on one doc or set of docs. So in your example please create a seperate folder and upload your doc "TGI.1001.pdf" in that folder. Now when doing search please select that folder from UI. This will allow you to focus on that document to generate answers without specifying in prompt. 

![image](https://github.com/microsoft/PubSec-Info-Assistant/assets/55892521/c3ce0f29-4114-4752-a2a6-3267f38e4655)
